library/modules/Gui.cpp

```diff
@@ -1360,6 +1362,137 @@ DFHACK_EXPORT void Gui::writeToGamelog(std::string message)
     fseed.close();
 }
 
+bool Gui::parseReportString(std::vector<std::string> &out, const std::string &str, size_t line_length)
+{   // out vector will contain strings cut to line_length, avoiding cutting up words
+    // Reverse-engineered from DF announcement code, fixes applied
```

Comment:
out of curiosity, what fixes were made?

Reply:
The issue with cutting the string at the wrong index, described in the notes of this bug report:
http://www.bay12games.com/dwarves/mantisbt/view.php?id=11813#c41165

Reply:
The unfixed code is posted on the Bay12 Forums:
http://www.bay12forums.com/smf/index.php?topic=98412.msg8324640#msg8324640